cleanlab/count.py

```diff
@@ -1108,12 +1111,31 @@ def get_confident_thresholds(labels: np.array, pred_probs: np.array) -> np.array
       class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
       higher) fold cross-validation.
 
+    multi_label : bool, optional
+      If ``True``, labels should be an iterable (e.g. list) of iterables, containing a
+      list of labels for each example, instead of just a single label.
+      The multi-label setting supports classification tasks where an example has 1 or more labels.
+      Example of a multi-labeled `labels` input: ``[[0,1], [1], [0,2], [0,1,2], [0], [1], ...]``.
+      The major difference in how this is calibrated versus single-label is that
+      the total number of errors considered is based on the number of labels,
+      not the number of examples. So, the calibrated `confident_joint` will sum
+      to the number of total labels.
+
     Returns
     -------
     confident_thresholds : np.array
       An array of shape ``(K,)``.
     """
-    confident_thresholds = np.array(
-        [np.mean(pred_probs[:, k][labels == k]) for k in range(pred_probs.shape[1])]
-    )
+
+    if multi_label:
+        # Compute unique number of classes K by flattening labels (list of lists)
+        K = len(np.unique([i for lst in labels for i in lst]))
```

Comment:
why not define `K = pred_probs.shape[1]` like is being done below when multi_label=false?

Reply:
otherwise I think anywhere num-classes are being computed based on num-unique-values, the docstring should explicitly state that all classes must be present in `labels` to avoid possible user confusion like we have seen in some Issues.

Reply:
overall I think it's important to be consistent more than anything else. So num_classes should be either be defined based on `pred_probs.shape[1]` or num-unique-values throughout the whole package. 

The former is more computationally efficient, but has the disadvantage that existing code would need to be changed (eg. in thresholds computation) to account for potentially missing classes not present in `labels`.

Reply:
Perf optimization if we use num-unique-values to set num-classes:  then certain helper functions could take in `num_classes` as an optional argument to avoid having to necessarily recompute: `unique(labels)` which may be costly for massive dataset.

Reply:
done